spec/integration/views_test_spec.rb

```diff
@@ -0,0 +1,151 @@
+    @comment2 = Comment.create(author_id: @profile1.id, post_id: @post1.id, Text: 'What is Up')
+    @like1 = Like.create(author_id: @profile1.id, post_id: @post1.id)
+  end
+  describe 'user index' do
+    before(:each) do
+      visit users_path
+    end
+    it 'Display the user details' do
+      expect(page).to have_content(@profile1.name)
+      expect(page).to have_content(@profile2.name)
+    end
+
+    it 'Display each user profile picture' do
+      expect(page).to have_xpath("//img[contains(@src,'https://as2.ftcdn.net/v2/jpg/03/26/98/51/1000_F_326985142_1aaKcEjMQW6ULp6oI9MYuv8lN9f8sFmj.jpg')]")
+    end
+
+    it 'Number of posts' do
+      expect(page).to have_content('Number of posts: 0')
+    end
+
+    it 'Redirect to user page' do
+      click_link 'Nabeel'
+      expect(page).to have_current_path("/users/#{@profile1.id}")
+    end
+  end
+
+  describe 'Show user' do
+    before(:each) do
+      visit user_path(@jin.id)
+    end
+    it 'Display each user name' do
+      expect(page).to have_content(@jin.name)
+      expect(page).to_not have_content(@profile2.name)
+    end
+    it 'Should show Number of posts by each user' do
+      expect(page).to have_content 'User Posts: 4'
+    end
+    it 'Should display user bio' do
+      expect(page).to have_content 'Teacher from Mexico.'
+    end
+    it 'I can see the user recent 3 posts.' do
+      expect(page).to_not have_content 'post content 4'
+    end
+    it 'I can see a button that show user all posts' do
+      expect(page).to have_content 'see more'
+    end
+    it 'When I click a users post, it redirects me to that posts show page.' do
+      visit "/users/#{@jin.id}/posts/#{@post1.id}"
+      expect(page).to have_content 'post content 1'
+    end
+    it 'When I click to see all posts, it redirects me to the users posts index page.' do
+      click_link 'see more'
+      expect(page).to have_content 'title 1'
+    end
+  end
+
+  describe 'post#index' do
+    before(:each) do
+      visit "/users/#{@jin.id}/posts"
+    end
+    it 'User name show' do
+      expect(page).to have_content @jin.name.to_s
+    end
+    it 'Number of posts user posted' do
+      expect(page).to have_content 'User Posts: 4'
+    end
+    it 'Post title' do
+      expect(page).to have_content 'title 1'
+    end
+    it 'Post body' do
+      expect(page).to have_content 'post content 1'
+    end
+    it 'Comments show' do
+      expect(page).to have_content 'Hello'
+    end
+    it 'comments count' do
+      expect(page).to have_content 'Comments: 2'
+    end
+    it 'likes count' do
+      expect(page).to have_content 'Likes: 1'
+    end
+    it 'click on link redirect to page' do
+      click_link 'title 1'
+      expect(page).to have_current_path("/users/#{@jin.id}/posts/#{@post1.id}")
+    end
+  end
+
+  describe 'Post#show' do
+    before(:each) do
+      visit "/users/#{@jin.id}/posts/#{@post1.id}"
+    end
```

Comment:
- [ ] According to project requirements it should be Tests that assert you can see the **post body** of all other users in **Post show page**. Please fix to make your test work as expected 😄